### PR TITLE
DBZ-7007 Use DebeziumEngine API instead of EmbeddedEngine in the testsuite

### PR DESCRIPTION
**Pull request by vjuranek**
https://issues.redhat.com/browse/DBZ-7007

### CODE REVIEW

**Comment by vjuranek** [comment]
MySQL 5.7 error is unrelated:
```
2023-10-25T11:44:01.4623251Z [ERROR] Errors: 
2023-10-25T11:44:01.4630678Z [ERROR] io.debezium.connector.mysql.MySqlParserIT.parseTableWithInVisibleColumns
2023-10-25T11:44:01.4632888Z [ERROR]   Run 1: MySqlParserIT.beforeEach:53 <C2><BB> Runtime Couldn't obtain stable version for image debezium/example-mysql
2023-10-25T11:44:01.4635184Z [ERROR]   Run 2: MySqlParserIT.afterEach:73 NullPointer Cannot invoke "org.testcontainers.containers.MySQLContainer.stop()" because "this.mySQLContainer" is null
2023-10-25T11:44:01.4636887Z [INFO] 
2023-10-25T11:44:01.4637470Z [INFO] 
2023-10-25T11:44:01.4641042Z [ERROR] Tests run: 317, Failures: 0, Errors: 1, Skipped: 10
```

DB2 failures are unrelated (`Db2ConnectorIT.restartInTheMiddleOfTx` and other known well-known failures)

Server error is unrelated and should be fixed by https://github.com/debezium/debezium-server/pull/44

However, Vitess error may be related.

**Comment by vjuranek** [comment]
It seems also Vitess failure is not related but a bug in the test - filed [DBZ-7104](https://issues.redhat.com/browse/DBZ-7104) for it.

**Comment by vjuranek** [comment]
DB2 failures are unrelated:
```
 Error:  Failures: 
Error:    Db2ConnectorIT.restartInTheMiddleOfTx:698->restartInTheMiddleOfTx:603 
Expected size:<30> but was:<0> in:
<[]>
Error:    Db2ConnectorIT.restartInTheMiddleOfTxAfterCompletedTx:692->restartInTheMiddleOfTx:603 
Expected size:<30> but was:<0> in:
<[]>
Error:    Db2ConnectorIT.restartInTheMiddleOfTxAfterSnapshot:686->restartInTheMiddleOfTx:603 
Expected size:<30> but was:<0> in:
<[]>
Error:    IncrementalSnapshotIT>AbstractIncrementalSnapshotTest.snapshotWithAdditionalConditionWithRestart:802->AbstractSnapshotTest.consumeMixedWithIncrementalSnapshot:233->AbstractSnapshotTest.consumeMixedWithIncrementalSnapshot:182->AbstractSnapshotTest.consumeMixedWithIncrementalSnapshot:200 [Too many no data record results, 0 < 1000] 
Expecting:
 <6>
to be less than or equal to:
 <5> 
Error:    IncrementalSnapshotIT>AbstractIncrementalSnapshotTest.snapshotWithAdditionalConditionWithSurrogateKey:847->AbstractConnectorTest.assertNoRecordsToConsume:984 expected:<[tru]e> but was:<[fals]e>
[INFO] 
Error:  Tests run: 61, Failures: 5, Errors: 0, Skipped: 2
```

File: debezium-embedded/src/test/java/io/debezium/embedded/AbstractConnectorTest.java [hunk 1]
```diff
@@ -101,10 +102,11 @@ public abstract class AbstractConnectorTest implements Testing {
     private static final String TEST_PROPERTY_PREFIX = "debezium.test.";
 
     private ExecutorService executor;
-    protected EmbeddedEngine engine;
+    protected TestingDebeziumEngine engine;
     protected BlockingQueue<SourceRecord> consumedLines;
     protected long pollTimeoutInMs = TimeUnit.SECONDS.toMillis(10);
     protected final Logger logger = LoggerFactory.getLogger(getClass());
+    protected final AtomicBoolean isRunning = new AtomicBoolean(false);
```
Comment (jpechane) [inline debezium-embedded/src/test/java/io/debezium/embedded/AbstractConnectorTest.java:109]: Please rename to `isEngineRunning` or to `isConnectorRunning` to make sure it is not confused with the notion of running test.

**Review by jpechane — COMMENTED** [review]
@vjuranek LGTM, thanks! I left one comment WRT renaming of the `isRunning` attribute. When done please feel free to merge PR yourself.

File: debezium-embedded/src/test/java/io/debezium/embedded/AbstractConnectorTest.java (hunk 1 above)
Comment (vjuranek) [reply debezium-embedded/src/test/java/io/debezium/embedded/AbstractConnectorTest.java]: done

**Comment by jpechane** [comment]
@vjuranek Applied, thanks

**Pull request merged**